crypto/objects/obj_dat.c

```diff
@@ -757,7 +757,7 @@ int OBJ_create(const char *oid, const char *sn, const char *ln)
 
     if (!ossl_obj_write_lock(1)) {
         ERR_raise(ERR_LIB_OBJ, ERR_R_UNABLE_TO_GET_WRITE_LOCK);
-        return 0;
+        goto err;
```

Comment:
You cannot do this here as there is an unlock call in err path. Which also means the `goto err;` above are wrong.